### PR TITLE
Feat: Add test file and implementation code for recipe repository class

### PR DESCRIPTION
**Pull request by tfettig22**
- What does this PR do: Write test and implementation code for recipe repository class
  - Also create smaller data set for testing purposes
- Where should reviewer start: start at Recipe-repository.js and Recipe-repository-test.js
- How to manually test: Test by running npm test in root directory
- Background info: Potential refactor of filter methods to use reduce rather than filter?

### CODE REVIEW

File: src/classes/RecipeRepository.js [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
+    return filteredRecipes;
+  }
+
+  filterByName(name) {
+    let filteredRecipes = this.recipeData.filter(recipe => recipe.name === name);
+    return filteredRecipes;
+  }
+
+  // try reduce later for the extention using multiple parameters? Maybe use ternary operator
+
```
Comment (GOECHA) [inline src/classes/RecipeRepository.js:21]: .filter() is a great method to filter through recipeData. I agree with the .reduce() iterator method to filter "tags", "recipe names", "ingredients" to return a recipe; for the extension. Thank you for submitting. CG

File: src/data/recipes.js [hunk 2]
```diff
@@ -5381,3 +5381,139 @@ const recipeData = [
       ]
     }
   ]
+
+
+  const recipeDataTest = [
```
Comment (GOECHA) [inline src/data/recipes.js:5386]: Noted, dataTest array for testing shorter data files.

File: test/RecipeRepository-test.js [hunk 3]
```diff
@@ -1,8 +1,38 @@
 import { expect } from 'chai';
 import RecipeRepository from '../src/classes/RecipeRepository';
+const data = require('../src/data/recipes.js')
+const testData = data.allRecipeData.testData
 
 describe('Recipe', () => {
+
+  let repository;
+
+  beforeEach(() => {
```
Comment (GOECHA) [inline test/RecipeRepository-test.js:10]: Great use of beforeEach() & ES6 - CG

**Review by GOECHA — APPROVED** [review]
Reviewed- updated data.

**Review by forsethnico — APPROVED** [review]
everything looks great! I agree with potentially updating the filter to reduce in future iterations.

**Pull request merged**